languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -185,28 +359,95 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   edge member -> def
 }
 
+(method_declaration (formal_parameters (_) @param)) @method
+{
+  edge @param.lexical_scope -> @method.lexical_scope
+  edge @method.lexical_scope -> @param.def
+  edge @method.defs -> @param.def
+}
+
+(formal_parameter type: (_) @type (_) @id) @param
+{
+  node @param.def
+  node @param.lexical_scope
+  node def__typeof
+
+  edge @type.lexical_scope -> @param.lexical_scope
+
+  attr (@param.def) node_definition = @id
+
+  attr (def__typeof) pop_symbol = ":"
+  edge def__typeof -> @type.type
+
+  edge @param.def -> def__typeof
+}
+
+(formal_parameter (modifiers) @modifiers) @this {
+  edge @modifiers.lexical_scope -> @this.lexical_scope
+}
+
+(method_declaration
+  (modifiers) @modifiers) @this {
+  edge @modifiers.lexical_scope -> @this.lexical_scope
+}
+
 (method_declaration
   body: (_) @stmt) @method
 {
   edge @stmt.before_scope -> @method.lexical_scope
 }
 
+(record_declaration name: (_) @name body: (_) @body) @this {
+  node def
+  attr (def) node_definition = @name
+  edge @this.defs -> def
```

Comment:
I think declarations like this should probably have `@this.def` as the interface to them, rather than `@this.defs`, since (AFAIK) no single declaration binds multiple names in the same scope. Future work.